### DISCUSSION TITLE
Error during the app creation using build api of codemagic.

### DISCUSSION THREAD

**Discussion by DeepakAphro121** [codemagic-ci-cd/codemagic-docs#2542] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

i got the issue when i was trying to build the new application using codemagic api then  app is created on admin but on admin we get an issue of "cannot access repository" . 

we follow the proper docs and use the proper ssh key still we get issue 

**Add a new application api**

  curl -H "Content-Type: application/json" \
       -H "x-auth-token: <API Token>" \
       -d '{
         "repositoryUrl": "git@github.com:my-organization/my-repo.git",
         "sshKey": {
           "data":  "MY_SSH_KEY", 
           "passphrase": null
         }
       }' \
       -X POST https://api.codemagic.io/apps

Issue i got
-----------

Cannot access the repository git@bitbucket.org:aphroecs/appscart_ios_test.git
Failed to fetch tags and branches with 'git ls-remote': Warning: Permanently added 'bitbucket.org,18.205.93.1' (ECDSA) to the list of known hosts. Load key "/tmp/tmp4l1g6z05": invalid format git@bitbucket.org: Permission denied (publickey). fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists. .

### Expected results

App should be created without getting any issue on admin of codemagic.

### Actual results

We are getting issue on app creation

Cannot access the repository git@bitbucket.org:aphroecs/appscart_ios_test.git
Failed to fetch tags and branches with 'git ls-remote': Warning: Permanently added 'bitbucket.org,18.205.93.1' (ECDSA) to the list of known hosts. Load key "/tmp/tmp4l1g6z05": invalid format git@bitbucket.org: Permission denied (publickey). fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists. .

**Comment by dtrdic** [comment]
Hey @DeepakAphro121 if your repositories are within an internal network / behind a firewall, please whitelist the following IP addresses through your firewall 
```
66.185.17.80/29 
199.7.162.128/29 
207.254.42.240/29 
34.74.234.56/32 
35.185.76.207/32
```

**Comment by DeepakAphro121** [comment]
we white listed all the api still we are not able to generate the app using codemagic api which is provided on codemagic docs diplay the unauthorized issue i provided proper ssh key 
 

API I USED 
  curl -H "Content-Type: application/json" \
       -H "x-auth-token: <API Token>" \
       -d '{
         "repositoryUrl": "git@github.com:my-organization/my-repo.git",
         "sshKey": {
           "data": "St89hgb-BASE64-ENCODED-SSH-KEY-FILE-H4ga7jgf==", 
           "passphrase": null
         }
       }' \
       -X POST https://api.codemagic.io/apps

**Comment by dtrdic** [comment, marked as answer]
Note: Issue resolved over the Discord channel. 
The user added the public key instead of the required private key to the API request

https://discord.com/channels/1131597315707261018/1242012994142212198/1243172549622235196
